crates/miden-testing/src/kernel_tests/tx/mod.rs

```diff
@@ -60,77 +68,119 @@ pub fn try_read_root_mem_word(process: &ProcessState, addr: u32) -> Option<Word>
+            ",
+        count = notes.len(),
+    ));
+
+    script
+}
 
-            # populate note 0
-            push.{note_0_metadata}
-            push.{OUTPUT_NOTE_SECTION_OFFSET}.{OUTPUT_NOTE_METADATA_OFFSET} add mem_storew dropw
-
-            push.{note_0_recipient}
-            push.{OUTPUT_NOTE_SECTION_OFFSET}.{OUTPUT_NOTE_RECIPIENT_OFFSET} add mem_storew dropw
-
-            push.{note_0_num_assets}
-            push.{OUTPUT_NOTE_SECTION_OFFSET}.{OUTPUT_NOTE_NUM_ASSETS_OFFSET} add mem_store
-
-            push.{}
-            push.{OUTPUT_NOTE_SECTION_OFFSET}.{OUTPUT_NOTE_ASSETS_OFFSET} add mem_storew dropw
-
-            # populate note 1
-            push.{note_1_metadata}
-            push.{OUTPUT_NOTE_SECTION_OFFSET}.{NOTE_1_OFFSET}.{OUTPUT_NOTE_METADATA_OFFSET} add add mem_storew dropw
-
-            push.{note_1_recipient}
-            push.{OUTPUT_NOTE_SECTION_OFFSET}.{NOTE_1_OFFSET}.{OUTPUT_NOTE_RECIPIENT_OFFSET} add add mem_storew dropw
-
-            push.{note_1_num_assets}
-            push.{OUTPUT_NOTE_SECTION_OFFSET}.{NOTE_1_OFFSET}.{OUTPUT_NOTE_NUM_ASSETS_OFFSET} add add mem_store
-
-            push.{}
-            push.{OUTPUT_NOTE_SECTION_OFFSET}.{NOTE_1_OFFSET}.{OUTPUT_NOTE_ASSETS_OFFSET} add add mem_storew dropw
-
-            # populate note 2
-            push.{note_2_metadata}
-            push.{OUTPUT_NOTE_SECTION_OFFSET}.{NOTE_2_OFFSET}.{OUTPUT_NOTE_METADATA_OFFSET} add add mem_storew dropw
-
-            push.{note_2_recipient}
-            push.{OUTPUT_NOTE_SECTION_OFFSET}.{NOTE_2_OFFSET}.{OUTPUT_NOTE_RECIPIENT_OFFSET} add add mem_storew dropw
+/// Creates a note with a note script that creates all `notes` that get passed as a parameter.
+///
+/// `note_asset` is the asset that the note itself will contain
+fn create_spawner_note(output_notes: Vec<&Note>) -> anyhow::Result<Note> {
+    create_spawner_note_with_assets(output_notes, vec![])
+}
 
-            push.{note_2_num_assets}
-            push.{OUTPUT_NOTE_SECTION_OFFSET}.{NOTE_2_OFFSET}.{OUTPUT_NOTE_NUM_ASSETS_OFFSET} add add mem_store
+/// Creates a note with a note script that creates all `notes` that get passed as a parameter,
+/// and that carries the passed `asset`.
+///
+/// `note_asset` is the asset that the note itself will contain
+fn create_spawner_note_with_assets(
+    output_notes: Vec<&Note>,
+    assets: Vec<Asset>,
+) -> anyhow::Result<Note> {
+    let note_code = note_script_that_creates_notes(output_notes);
+
+    let note = NoteBuilder::new(ACCOUNT_ID_SENDER.try_into()?, rng())
+        .code(note_code)
+        .add_assets(assets)
+        .build(&TransactionKernel::testing_assembler_with_mock_account())
+        .unwrap();
+
+    Ok(note)
+}
 
-            push.{}
-            push.{OUTPUT_NOTE_SECTION_OFFSET}.{NOTE_2_OFFSET}.{OUTPUT_NOTE_ASSETS_OFFSET} add add mem_storew dropw
+/// Returns the code for a note that creates all notes in `output_notes`
+fn note_script_that_creates_notes(output_notes: Vec<&Note>) -> String {
```

Comment:
This is the other useful function that I factored out. It was very hardcoded before for a specific amount of output notes but this generalizes it a little bit also.